spec/models/spree/adjustment_spec.rb

```diff
@@ -212,54 +218,114 @@ module Spree
 
-          it "records 0% tax on shipment adjustments" do
-            allow(Config).to receive(:shipping_tax_rate).and_return(0)
-            order.shipments = [shipment]
-
-            expect(order.shipment_adjustments.first.included_tax).to eq(0)
+          context "when the shipment has an inclusive tax rate" do
+            it "calculates the shipment tax from the tax rate" do
+              order.shipments = [shipment]
+              order.create_tax_charge!
+              order.update_totals
+
+              # Finding the tax included in an amount that's already inclusive of tax:
+              # total - ( total / (1 + rate) )
+              # 50    - ( 50    / (1 + 0.25) )
+              # = 10
+              expect(order.shipment_adjustments.tax.first.amount).to eq(10)
+              expect(order.shipment_adjustments.tax.first.included).to eq true
+
+              expect(shipment.reload.cost).to eq(50)
+              expect(shipment.included_tax_total).to eq(10)
+              expect(shipment.additional_tax_total).to eq(0)
+
+              expect(order.included_tax_total).to eq(10)
+              expect(order.additional_tax_total).to eq(0)
+            end
           end
 
-          it "records 0% tax on shipments when a rate is set but shipment_inc_vat is false" do
-            allow(Config).to receive(:shipping_tax_rate).and_return(0.25)
-            order.shipments = [shipment]
+          context "when the shipment has an added tax rate" do
+            let(:inclusive_tax) { false }
 
-            expect(order.shipment_adjustments.first.included_tax).to eq(0)
-          end
-        end
+            # Current behaviour. Will be replaced by the pending test below
+            it "records the tax on the order's adjustments" do
+              order.shipments = [shipment]
+              order.create_tax_charge!
+              order.update_totals
 
-        describe "when tax on shipping is enabled" do
-          before do
-            allow(Config).to receive(:shipment_inc_vat).and_return(true)
-          end
+              expect(order.shipment_adjustments.tax.count).to be_zero
 
-          it "takes the shipment adjustment tax included from the system setting" do
-            allow(Config).to receive(:shipping_tax_rate).and_return(0.25)
-            order.shipments = [shipment]
+              # Finding the added tax for an amount:
+              # total * rate
+              # 50    * 0.25
+              # = 12.5
+              expect(order.adjustments.tax.first.amount).to eq(12.50)
+              expect(order.adjustments.tax.first.included).to eq false
+
+              expect(shipment.reload.cost).to eq(50)
+              expect(shipment.included_tax_total).to eq(0)
+              expect(shipment.additional_tax_total).to eq(0)
+
+              expect(order.included_tax_total).to eq(0)
+              expect(order.additional_tax_total).to eq(12.50)
+            end
+
+            xit "records the tax on the shipment's adjustments" do
```

Comment:
is the `xit` on purpose?

Reply:
In this case yes. It's for additional taxes, and the behavior is still a bit weird until we finish addressing them. This test will be used soon, and the one above will be deleted :+1: